src/Lexicographics.dfy

```diff
@@ -6,90 +6,109 @@
+    assert Relations.Transitive(ByteSeqBelow) by {
       forall x, y, z, n: nat |
         n <= |x| && n <= |y| && n <= |z| &&
-        LexicographicByteSeqBelowAux(x, y, n) && LexicographicByteSeqBelowAux(y, z, n)
+        ByteSeqBelowAux(x, y, n) && ByteSeqBelowAux(y, z, n)
       {
-        AboutLexicographicByteSeqBelowAux_Transitive(x, y, z, n);
+        AboutByteSeqBelowAux_Transitive(x, y, z, n);
       }
     }
-    assert Connected(LexicographicByteSeqBelow) by {
+    assert Connected(ByteSeqBelow) by {
       forall x, y, n: nat | n <= |x| && n <= |y| {
-        AboutLexicographicByteSeqBelowAux_Connected(x, y, n);
+        AboutByteSeqBelowAux_Connected(x, y, n);
       }
     }
   }
 
-  lemma AboutLexicographicByteSeqBelowAux_Reflexive(x: seq<uint8>, n: nat)
+  lemma AboutByteSeqBelowAux_Reflexive(x: seq<uint8>, n: nat)
     requires n <= |x|
-    ensures LexicographicByteSeqBelowAux(x, x, n)
+    ensures ByteSeqBelowAux(x, x, n)
     decreases |x| - n
   {
   }
 
-  lemma AboutLexicographicByteSeqBelowAux_AntiSymmetric(x: seq<uint8>, y: seq<uint8>, n: nat)
+  lemma AboutByteSeqBelowAux_AntiSymmetric(x: seq<uint8>, y: seq<uint8>, n: nat)
     requires n <= |x| && n <= |y|
     requires x[..n] == y[..n]
-    requires LexicographicByteSeqBelowAux(x, y, n) && LexicographicByteSeqBelowAux(y, x, n)
+    requires ByteSeqBelowAux(x, y, n) && ByteSeqBelowAux(y, x, n)
     ensures x == y
     decreases |x| - n
   {
   }
 
-  lemma AboutLexicographicByteSeqBelowAux_Transitive(x: seq<uint8>, y: seq<uint8>, z: seq<uint8>, n: nat)
+  lemma AboutByteSeqBelowAux_Transitive(x: seq<uint8>, y: seq<uint8>, z: seq<uint8>, n: nat)
     requires n <= |x| && n <= |y| && n <= |z|
-    requires LexicographicByteSeqBelowAux(x, y, n) && LexicographicByteSeqBelowAux(y, z, n)
-    ensures LexicographicByteSeqBelowAux(x, z, n)
+    requires ByteSeqBelowAux(x, y, n) && ByteSeqBelowAux(y, z, n)
+    ensures ByteSeqBelowAux(x, z, n)
     decreases |x| - n
   {
   }
 
-  lemma AboutLexicographicByteSeqBelowAux_Connected(x: seq<uint8>, y: seq<uint8>, n: nat)
+  lemma AboutByteSeqBelowAux_Connected(x: seq<uint8>, y: seq<uint8>, n: nat)
     requires n <= |x| && n <= |y|
-    ensures LexicographicByteSeqBelowAux(x, y, n) || LexicographicByteSeqBelowAux(y, x, n)
+    ensures ByteSeqBelowAux(x, y, n) || ByteSeqBelowAux(y, x, n)
     decreases |x| - n
   {
   }
+
+  /*
+   * Lexicographic comparison of sequences.
+   *
+   * Given two sequences `a` and `b` and a strict (that is, irreflexive)
+   * ordering `less` on the elements of these sequences, determine whether or not
+   * `a` is lexicographically "less than or equal to" `b`.
+   *
+   * `a` is lexicographically "less than or equal to" `b` holds iff
+   *   there exists a `k` such that
+   *   - the first `k` elements of `a` and `b` are the same
+   *   - either:
+   *      -- `a` has length `k` (that is, `a` is a prefix of `b`)
+   *      -- `a[k]` is strictly less (using `less`) than `b[k]`
+   */
```

Comment:
This comment should be moved to the right module.

Reply:
I will rename "Lexicographics" to a better describing name ,I just wanted to make sure there is no other file changes before doing that.